File: telerik_ui_for_winforms_courseware_chm/Courseware/Treeview/CS/DatabindingTreeView/Databinding/XMLSerialize.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.Primitives;
using Telerik.WinControls.UI;
using System.IO;

namespace Databinding
{
    public partial class XMLSerialize : RadForm
    {
        public XMLSerialize()
        {
            InitializeComponent();
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            SaveFileDialog dialog = new SaveFileDialog();
            dialog.Filter = "XML Files|*.xml";
            if (dialog.ShowDialog() == DialogResult.OK)
            {
                radTreeView1.SaveXML(dialog.FileName);
            }
        }

        private void btnLoad_Click(object sender, EventArgs e)
        {
            OpenFileDialog dialog = new OpenFileDialog();
            dialog.Filter = "XML Files|*.xml";
            if (dialog.ShowDialog() == DialogResult.OK)
            {
                radTreeView1.LoadXML(dialog.FileName);
            }
        }

        private void btnAddSibling_Click(object sender, EventArgs e)
        {
            // get the sibling level nodes collection and add to it
            RadTreeNodeCollection nodes = null;
            if ((radTreeView1.SelectedNode == null) ||
                (radTreeView1.SelectedNode.Parent == null))
                nodes = radTreeView1.Nodes;
            else
            {
                nodes = radTreeView1.SelectedNode.Parent.Nodes;
            }
            RadTreeNode node = new RadTreeNode(tbNodeText.Text);
            nodes.Add(node);
            radTreeView1.BringIntoView(node);

        }

        private void btnAddChild_Click(object sender, EventArgs e)
        {
            // add a node directly to the selected node
            if (radTreeView1.SelectedNode != null)
            {
                RadTreeNode node = new RadTreeNode(tbNodeText.Text);
                radTreeView1.SelectedNode.Nodes.Add(node);
                radTreeView1.BringIntoView(node);
            }
        }

        private void btnDeleteNode_Click(object sender, EventArgs e)
        {
            // delete the selected node
            if (radTreeView1.SelectedNode != null)
                radTreeView1.SelectedNode.Remove();
        }
    }
}

File: telerik_ui_for_winforms_courseware_chm/Courseware/Treeview/VB/DatabindingTreeView/Databinding/XMLSerialize.vb
Imports Telerik.WinControls.UI

Public Class XMLSerialize
    Private Sub btnSave_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnSave.Click
        Dim dialog As SaveFileDialog = New SaveFileDialog()
        dialog.Filter = "XML Files|*.xml"
        If dialog.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
            RadTreeView1.SaveXML(dialog.FileName)
        End If
    End Sub

    Private Sub btnLoad_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnLoad.Click
        Dim dialog As OpenFileDialog = New OpenFileDialog()
        dialog.Filter = "XML Files|*.xml"
        If dialog.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
            RadTreeView1.LoadXML(dialog.FileName)
        End If
    End Sub

    Private Sub btnAddSibling_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnAddSibling.Click
        ' get the sibling level nodes collection and add to it
        Dim nodes As RadTreeNodeCollection = Nothing
        If (RadTreeView1.SelectedNode Is Nothing) OrElse (RadTreeView1.SelectedNode.Parent Is Nothing) Then
            nodes = RadTreeView1.Nodes
        Else
            nodes = RadTreeView1.SelectedNode.Parent.Nodes
        End If
        Dim node As New RadTreeNode(tbNodeText.Text)
        nodes.Add(node)
        RadTreeView1.BringIntoView(node)


    End Sub

    Private Sub btnAddChild_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnAddChild.Click
        ' add a node directly to the selected node
        If Not RadTreeView1.SelectedNode Is Nothing Then
            Dim node As New RadTreeNode(tbNodeText.Text)
            RadTreeView1.SelectedNode.Nodes.Add(node)
            RadTreeView1.BringIntoView(node)
        End If

    End Sub

    Private Sub btnDelete_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnDeleteNode.Click
        ' delete the selected node
        If Not RadTreeView1.SelectedNode Is Nothing Then
            RadTreeView1.SelectedNode.Remove()
        End If
    End Sub

End Class